assets/js/googlesitekit/datastore/site/consent-mode.js

```diff
@@ -164,6 +166,30 @@ const baseSelectors = {
 	getConsentAPIInfo: ( state ) => {
 		return state.consentMode.apiInfo;
 	},
+
+	/**
+	 * Returns true if Google Ads is in use, either through a linked Analytics & Ads account
+	 * or an Ads conversion tracking ID.
+	 *
+	 * TODO: The Ads conversion tracking ID is being moved to the new "Ads" module. Source
+	 * this ID from the "Ads" module once it's implemented.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} True if Google Ads is in use, false otherwise.
+	 */
+	isAdsConnected: createRegistrySelector( ( select ) => () => {
+		const { isModuleConnected } = select( CORE_MODULES );
+
+		if ( ! isModuleConnected( 'analytics-4' ) ) {
+			return false;
+		}
+
+		const { getAdsConversionID, getAdsLinked } =
+			select( MODULES_ANALYTICS_4 );
+
+		return !! getAdsConversionID() || !! getAdsLinked();
```

Comment:
Oh, one question I just thought of: should we be returning `undefined` from here if these selectors are still loading? Right now `undefined` would return `false`, but that might cause this to go from a "false" to "true" state during loading, which isn't expected. 🤔

I don't think it affects any current components, but it might trip us up in the future. Can you add some `undefined` handling here? (Unless it isn't needed.)

Reply:
I agree. I have updated the selector to return `undefined` if any of these two selectors return `undefined`.

You're right. I also don't think this affects any current components, as a result, I haven't added any loading state into them. But please let me know if I should do so instead.

Please let me know if this looks good now, thank you!